python/cudf/cudf/core/column_accessor.py

```diff
@@ -341,6 +341,84 @@ def _pad_key(self, key: Any, pad_value="") -> Any:
             key = (key,)
         return key + (pad_value,) * (self.nlevels - len(key))
 
+    def replace_level_values(
```

Comment:
Do we need to implement this all the way down at the `ColumnAccessor`? This feels like something we should be able to accomplish in `Frame` instead.

Reply:
It was my suggestion initially to move the logic here. But after giving it some thought I think I agree with Keith here.

ColumnAccessor was meant initially to be really a dict-like object but it has grown into something much more than that.

Likely we want to move this (and other methods in this class) into either free functions that operate on and return CAs, or directly into the Frame class.

Reply:
Even if we make them free methods we should have them operate on Frames instead of ColumnAccessor objects whenever possible.

Reply:
That just makes those methods do too much, since a rename is exclusively an operation on the keys of a column accessor.

Reply:
So my suggestion would be something like:

```python
def rename_levels(ca: ColumnAccessor, levels, mapper):
     """Rename the specified levels of the given ColumnAccessor"""
    pass

class Frame:
     def rename(self, ...):
          new_data = rename_levels(self._data, ...)
          return self.__class__._from_data(new_data)
```

I think the above separates concerns a little better than implementing _all_ the logic in `Frame.rename`. Conceivably, renaming the levels of a ColumnAccessor could be useful outside of `Frame.rename`.

I also think it's better to write functions that operate directly on ColumnAccessors rather than Frames, as there might be situations where we want to compose ColumnAccessor operations  (e.g., insert_level + rename).

Reply:
Fair enough, lets move forward with it as a ColumnAccessor function